public/app/features/dashboard/dashboard_ctrl.ts

```diff
@@ -29,11 +30,15 @@ export class DashboardCtrl {
       var resizeEventTimeout;
 
       $scope.setupDashboard = function(data) {
-        var dashboard = dashboardSrv.create(data.dashboard, data.meta);
-        dashboardSrv.setCurrent(dashboard);
-
-        // init services
-        timeSrv.init(dashboard);
+        try {
+          var dashboard = dashboardSrv.create(data.dashboard, data.meta);
+          dashboardSrv.setCurrent(dashboard);
+
+          // init services
+          timeSrv.init(dashboard);
+        } catch (err) {
+          $scope.appEvent("alert-error", ['Dashboard init failed', err.message]);
```

Comment:
this does not look right will just continue down the function and cause other errors? Maybe combine this exception handler with the one below ?